src/main/scala/is/hail/distributedmatrix/BlockMatrixIsDistributedMatrix.scala

```diff
@@ -0,0 +1,177 @@
+package is.hail.distributedmatrix
+
+import is.hail.utils._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.mllib.linalg._
+import org.apache.spark.mllib.linalg.distributed._
+
+import scala.reflect.ClassTag
+
+object BlockMatrixIsDistributedMatrix extends DistributedMatrix[BlockMatrix] {
+  type M = BlockMatrix
+
+  def cache(m: M): M = m.cache()
+
+  def from(rdd: RDD[Array[Double]]): M =
+    new IndexedRowMatrix(rdd.zipWithIndex().map { case (x, i) => new IndexedRow(i, new DenseVector(x)) })
+      .toBlockMatrixDense()
+  def from(bm: BlockMatrix): M = bm
+  def from(cm: CoordinateMatrix): M = cm.toBlockMatrix()
+
+  def transpose(m: M): M = m.transpose
+  def diagonal(m: M): Array[Double] = toCoordinateMatrix(m)
+    .entries
+    .filter(me => me.i == me.j)
+    .map(me => (me.i, me.value))
+    .collect()
+    .sortBy(_._1)
+    .map(_._2)
+
+  def multiply(l: M, r: M): M = l.multiply(r)
+  def multiply(l: M, r: DenseMatrix): M = {
+    val sc = l.blocks.sparkContext
+    val rbc = sc.broadcast(r)
```

Comment:
Should we do a dimensions check before doing all of this? The eventual call to l.multiply will catch it, but we do whatever work is in this function for no reason if the dimensions are wrong.